tests/utils/streams/test_batching.py

```diff
@@ -1,31 +1,66 @@
 import time
 from datetime import datetime
-from typing import Any, MutableSequence, Sequence, Optional
+from typing import Any, Callable, Mapping, MutableMapping, MutableSequence, Sequence, Optional
 from unittest.mock import patch
 
-from confluent_kafka import Message
-
 from snuba.utils.metrics.backends.dummy import DummyMetricsBackend
+from snuba.utils.streams.abstract import Consumer
 from snuba.utils.streams.batching import AbstractBatchWorker, BatchingKafkaConsumer
-from tests.backends.confluent_kafka import FakeConfluentKafkaConsumer, FakeConfluentKafkaMessage, FakeConfluentKafkaProducer
+from snuba.utils.streams.kafka import KafkaMessage, TopicPartition
+from tests.backends.confluent_kafka import FakeConfluentKafkaProducer
+
+
+class FakeKafkaConsumer(Consumer[TopicPartition, int, bytes]):
+    def __init__(self):
+        self.items: MutableSequence[KafkaMessage] = []
+        self.commit_calls = 0
+        self.close_calls = 0
+        self.positions: MutableMapping[TopicPartition, int] = {}
+
+    def subscribe(
+        self,
+        topics: Sequence[str],
+        on_assign: Optional[Callable[[Sequence[TopicPartition]], None]] = None,
+        on_revoke: Optional[Callable[[Sequence[TopicPartition]], None]] = None,
+    ) -> None:
+        pass  # XXX: This is a bit of a smell.
+
+    def poll(
+        self, timeout: Optional[float] = None
+    ) -> Optional[KafkaMessage]:
+        try:
+            message = self.items.pop(0)
+        except IndexError:
+            return None
+
+        self.positions[message.stream] = message.offset + 1
+
+        return message
+
+    def commit(self) -> Mapping[TopicPartition, int]:
+        self.commit_calls += 1
+        return self.positions
+
+    def close(self) -> None:
+        self.close_calls += 1
```

Comment:
This is basically a port of `tests.backends.confluent_kafka.FakeConfluentKafkaConsumer` that uses the new types. This will eventually be be made more generic as well and will become a `FakeConsumer` and probably get moved into `test.backends.streams` or something instead.